scala/scala.js

```diff
@@ -0,0 +1,1270 @@
+      ready(instance.find('.jsvee-value').first().clone());
+    }
+  };
+
+  JSVEE.handlers.classes['Category'] = function (ready, area, element) {
+
+    if (element.hasClass("jsvee-function") && element.attr('data-name') == 'Category') {
+
+      var id = $(element.find('.jsvee-value').get(0)).attr('data-id');
+      var instance = area.find('.jsvee-heap .jsvee-instance[data-id="' + id + '"]').first();
+      instance.attr('data-best', '-1');
+      instance.attr('data-bestRef', '-1');
+      var names = [ "name", "unit" ];
+
+      for ( var i = 0; i < names.length; i++) {
+        var value = $(element.find('.jsvee-value').get(i + 1));
+        var name = names[i];
+        var variable = JSVEE.utils.ui.createVariable(name);
+        variable.append(value.clone());
+        instance.append(variable);
+      }
+
+      ready(JSVEE.utils.ui.createReference(area, id));
+
+    }
+
+    if (element.hasClass("jsvee-function") && element.attr('data-name') == 'addExperience') {
+      var id = $(element.find('.jsvee-value').get(0)).attr('data-id');
+      var instance = area.find('.jsvee-heap .jsvee-instance[data-id="' + id + '"]').first();
+
+      var expId = $(element.find('.jsvee-value').get(1)).data('id');
+      var expInstance = area.find('.jsvee-heap .jsvee-instance[data-id="' + expId + '"]').first();
+
+      var other = expInstance.find('.jsvee-variable[data-name="rating"] .jsvee-value').first().attr('data-value');
+
+      if (parseInt(instance.attr('data-best'), 10) < parseInt(other, 10)) {
+        instance.attr('data-best', other);
+        instance.attr('data-bestRef', expId);
+      }
+
+      ready(createUnit());
+
+    }
+
+    if (element.hasClass("jsvee-function") && element.attr('data-name') == 'favorite') {
+      var id = $(element.find('.jsvee-value').get(0)).attr('data-id');
+      var instance = area.find('.jsvee-heap .jsvee-instance[data-id="' + id + '"]').first();
+
+      var bestRef = instance.attr('data-bestRef');
+
+      ready(JSVEE.utils.ui.createReference(this.area, bestRef));
+    }
+
+  };
+
+  JSVEE.handlers.classes['Experience'] = function (ready, area, element) {
+
+    if (element.hasClass("jsvee-function") && element.attr('data-name') == 'Experience') {
+
+      var id = $(element.find('.jsvee-value').get(0)).attr('data-id');
+      var instance = area.find('.jsvee-heap .jsvee-instance[data-id="' + id + '"]').first();
+
+      var names = [ "name", "description", "price", "rating" ];
+
+      for ( var i = 0; i < names.length; i++) {
+        var value = $(element.find('.jsvee-value').get(i + 1));
+        var name = names[i];
+        var variable = JSVEE.utils.ui.createVariable(name);
+        variable.append(value.clone());
+        instance.append(variable);
+      }
+
+      ready(JSVEE.utils.ui.createReference(area, id));
+    }
+
+  };
+
+  JSVEE.handlers.classes['EnglishAuction'] = function (ready, area, element) {

```

Comment:
These course specific classes should probably be split to another js file that is included per course.